tls/s2n_config.c

```diff
@@ -65,17 +65,40 @@ static int wall_clock(void *data, uint64_t *nanoseconds)
 
 static struct s2n_config s2n_default_config = { 0 };
 static struct s2n_config s2n_default_fips_config = { 0 };
-static struct s2n_config s2n_default_tls13_config = { 0 };
+/* Dedicated configs for testing different protocols since tests can override the
+ * default security policy behavior.
+ */
+static struct s2n_config s2n_testing_default_tls12_config = { 0 };
+static struct s2n_config s2n_testing_default_tls12_fips_config = { 0 };
+static struct s2n_config s2n_testing_default_tls13_config = { 0 };
```

Comment:
Do these need to live in the main source? Or can they be moved to the testing library? You're adding more complexity to the source here than I think you need to. See if you can simplify.

("default" and "default_fips" have real customer use cases. "default_tls13" had real customer use cases when it was first added, before we changed how we configured tls1.3, but could probably be removed now)